.github/workflows/ci.yml

```
@@ -0,0 +1,103 @@
+name: CI
+
+env: 
+  IMAGE_NAME: activemq-artemis-self-provisioning-plugin
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+  workflow_dispatch:
+
+jobs:
+  analyze:
+    name: Analyze
+    runs-on: ubuntu-latest
+    permissions:
+      actions: read
+      contents: read
+      security-events: write
+
+    strategy:
+      fail-fast: false
+      matrix:
+        language: ["javascript"]
+        # CodeQL supports [ 'cpp', 'csharp', 'go', 'java', 'javascript', 'python', 'ruby' ]
+        # Learn more about CodeQL language support at https://aka.ms/codeql-docs/language-support
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+
+      # Initializes the CodeQL tools for scanning.
+      - name: Initialize CodeQL
+        uses: github/codeql-action/init@v2
+        with:
+          languages: ${{ matrix.language }}
+          
+      - name: Perform CodeQL Analysis
+        uses: github/codeql-action/analyze@v2
+
+  build:
+    name: Build
+    runs-on: ubuntu-latest
+    steps:
+
+      - name: Checkout the repo
+        uses: actions/checkout@v3
+
+      - uses: actions/setup-node@v3
+        with:
+          node-version: 16.6.2
+
+      - name: Get node version
+        id: node
+        run: |
+            echo ":: set-output name=version::$(node -v)"
+
+      - name: Get node_modules cache
+        uses: actions/cache@v3
+        id: yarn-cache
+        with:
+          path: |
+              ~/.yarn
+              **/node_modules
+          key: ${{ runner.os }}-yarn-${{ hashFiles('**/yarn.lock') }}-${{ steps.node.outputs.version }}
+
+      - run: yarn install --frozen-lockfile
```

```suggestion
     -  name: Install dependencies
        if: steps.node-cache.outputs.cache-hit != 'true'
        shell: bash
        run: yarn install --frozen-lockfile
```

@AjayJagan Let's keep existing separate cache file i.e. `actions/cache-restore/action.yml` and modify if any changes required.  There are some check missing here for cache like  if: steps.node-cache.outputs.cache-hit != 'true'.